rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -207,7 +185,103 @@ public void getRecoveringLRAs(Map<URI, Transaction> lras) {
+        InputObjectState aa_uids = new InputObjectState();
+        AtomicBoolean success = new AtomicBoolean(false);
+        Consumer<Uid> uidDestroyer = uid -> {
+            Transaction lra = new Transaction(lraService, new Uid(uid));
+
+            lra.activate(); // or just parse lraId extracting the uid
+
+            if (uid.fileStringForm().equals(lra.getUid())) {
+                try {
+                    success.set(_recoveryStore.remove_committed(uid, lra.type()));
+                } catch (ObjectStoreException e) {
+                    if (LRALogger.logger.isTraceEnabled()) {
+                        LRALogger.logger.tracef(
+                                "LRARecoverModule: Object store exception while removing record %s",
+                                uid.fileStringForm(), e);
+                    }
+                }
+            }
+        };
+
+        if (getUids(aa_uids)) {
+            forEach(aa_uids, uidDestroyer);
+        }
+
+        return success.get();
+    }
+
+    public void getFailedLRAs(Map<URI, Transaction> lras) {
+        InputObjectState aa_uids = new InputObjectState();
+        Consumer<Uid> failedLRACreator = uid -> {
+            Transaction lra = new Transaction(lraService, new Uid(uid));
+
+            lra.activate();
+            lras.put(lra.getId(), lra);
+        };
+
+        if (getUids(aa_uids)) {
+            forEach(aa_uids, failedLRACreator);
+        }
+    }
+
+    private boolean getUids(InputObjectState aa_uids) {
+        synchronized (this) {
+            try {
+                return _recoveryStore.allObjUids(_transactionType, aa_uids);
+            } catch (ObjectStoreException e) {
+                if (LRALogger.logger.isTraceEnabled()) {
+                    LRALogger.logger.tracef(
+                            "LRARecoverModule: Exception %s while unpacking records of type %s",
+                            e.getMessage(), _transactionType, e);
+                } else if (LRALogger.logger.isInfoEnabled()) {
+                    LRALogger.logger.infof("LRARecoverModule: Exception %s while unpacking records of type : %s",
+                            e.getMessage());
+                }
+
+                return false;
+
+            }
+        }
+    }
+
+    /**
+     * Iterate over a collection of Uids
+     * This could be moved to an ArjunaCore class (such as UidHelper) if its useful
+     *
+     * @param uids the uids to iterate over
+     * @param consumer the consumer that should be called for each Uid
+     */
+    private void forEach(InputObjectState uids, Consumer<Uid> consumer) {
+        do {
+            try {
+                Uid uid = new Uid(uids.unpackBytes());
+
+                if (uid.equals(Uid.nullUid())) {
+                    return;
+                }
+
+                consumer.accept(uid);
+            } catch (IOException e) {
+                if (LRALogger.logger.isTraceEnabled()) {
+                    LRALogger.logger.tracef(
```

Comment:
the same as above, please use the `tracef(Throwable t, String format, Object... params)`

Reply:
fixed with the latest push